### PR TITLE
catalog-model: update ownership of data

### PR DESCRIPTION
**Pull request by freben**
Now the guest user owns some stuff again.

### CODE REVIEW

File: plugins/catalog/src/components/useOwnUser.ts [hunk 1]
```diff
@@ -21,20 +21,21 @@ import { identityApiRef, useApi } from '@backstage/core';
 import { catalogApiRef } from '../plugin';
 
 /**
- * Get the group memberships of the logged-in user.
+ * Get the catalog User entity (if any) that matches the logged-in user.
  */
-export function useOwnUser(): AsyncState<UserEntity> {
+export function useOwnUser(): AsyncState<UserEntity | undefined> {
```
Comment (freben) [inline plugins/catalog/src/components/useOwnUser.ts:26]: this is not a functional change nor used outside of the package, so needs no changeset

**Review by Fox32 — APPROVED** [review]
👍 Even includes a user-as-owner scenario, cool!

**Comment by lewtakm** [comment]
Nice, was just planning to create lots of different yaml files to register components with proper ownerships field in order to test connections between groups and components. Now it's out of the box 👍

**Review by benjdlambert — APPROVED** [review]

**Comment by freben** [comment]
@lewtakt now!

**Pull request merged**